python/iceberg/api/expressions/projections.py

```diff
@@ -15,32 +15,81 @@
 # specific language governing permissions and limitations
 # under the License.
 
+from typing import TYPE_CHECKING
 
 from .expressions import Expressions, ExpressionVisitors, RewriteNot
 from .predicate import BoundPredicate, UnboundPredicate
 
+if TYPE_CHECKING:
+    from .expression import Expression
+    from .predicate import Predicate
+    from ..partition_spec import PartitionSpec
 
-def inclusive(spec, case_sensitive=True):
+
```

Comment:
it seems a bit weird to me to expose `inclusive` and `InclusiveProjection` in the public API of this module. I wonder if we should remove `InclusiveProjection` from `__init__.py` of this module to ensure everyone uses the well documented `inclusive` function. (same for `strict` below)

Reply:
agree, I'll update this